packages/core/src/types.ts

```diff
@@ -18,24 +18,22 @@ export interface IPrismConfig {
 export type ValidatorFn<Resource, T> = (opts: { resource: Resource; element: T }) => IPrismDiagnostic[];
 
 export type IPrismComponents<Resource, Input, Output, Config extends IPrismConfig> = {
-  route: (opts: { resources: Resource[]; input: Input; config?: Config }) => Either<Error, Resource>;
-  validateInput?: ValidatorFn<Resource, Input>;
-  validateOutput?: ValidatorFn<Resource, Output>;
+  route: (opts: { resources: Resource[]; input: Input }) => Either<Error, Resource>;
+  validateInput: ValidatorFn<Resource, Input>;
+  validateOutput: ValidatorFn<Resource, Output>;
   mock: (
     opts: {
       resource: Resource;
       input: IPrismInput<Input>;
-      config?: Config;
+      config: Config['mock'];
```

Comment:
Correct, but on the other hand we figured that every optional parameter we take would be another `Option<T>` to handle and so, if not necessary, we prefer to pass all the parameters explicitly.

Plus, this makes much easier to chain them with a `Reader` in case it's going to be needed on day.